src/earn/EarnEnterAmount.tsx

```diff
@@ -282,45 +325,69 @@ function EarnEnterAmount({ route }: Props) {
     // eventually we may want to do something smarter here, like subtracting gas fees from the max amount if
     // this is a gas-paying token. for now, we are just showing a warning to the user prompting them to lower the amount
     // if there is not enough for gas
-    setTokenAmountInput(token.balance.toFormat({ decimalSeparator }))
+    if (isWithdrawal) {
+      setTokenAmountInput(balanceInTransactionToken.toFormat({ decimalSeparator }))
+    } else {
+      setTokenAmountInput(inputToken.balance.toFormat({ decimalSeparator }))
+    }
     setEnteredIn('token')
+    setMaxPressed(true)
     tokenAmountInputRef.current?.blur()
     localAmountInputRef.current?.blur()
     AppAnalytics.track(SendEvents.max_pressed, {
-      tokenId: token.tokenId,
-      tokenAddress: token.address,
-      networkId: token.networkId,
+      tokenId: transactionToken.tokenId,
+      tokenAddress: transactionToken.address,
+      networkId: transactionToken.networkId,
+      mode,
     })
   }
 
   const onPressContinue = () => {
-    if (!tokenAmount || !depositToken) {
+    if (!tokenAmount || !transactionToken) {
       // should never happen
       return
     }
     AppAnalytics.track(EarnEvents.earn_enter_amount_continue_press, {
-      amountInUsd: tokenAmount.multipliedBy(token.priceUsd ?? 0).toFixed(2),
+      amountInUsd: isWithdrawal
+        ? tokenAmount.multipliedBy(inputToken.priceUsd ?? 0).toFixed(2)
+        : tokenAmount.multipliedBy(transactionToken.priceUsd ?? 0).toFixed(2),
       amountEnteredIn: enteredIn,
       depositTokenId: pool.dataProps.depositTokenId,
-      networkId: token.networkId,
+      networkId: transactionToken.networkId,
       providerId: pool.appId,
       poolId: pool.positionId,
-      fromTokenId: token.tokenId,
+      fromTokenId: transactionToken.tokenId,
       fromTokenAmount: tokenAmount.toString(),
       mode,
       depositTokenAmount: swapTransaction
         ? getSwapToAmountInDecimals({ swapTransaction, fromAmount: tokenAmount }).toString()
         : tokenAmount.toString(),
     })
+    // TODO(ACT-1389) if isWithdrawal === true navigate to EarnConfirmationScreen
     reviewBottomSheetRef.current?.snapToIndex(0)
   }
 
+  const dropdownEnabled = useMemo(() => {
+    switch (mode) {
+      case 'withdraw':
+        return withdrawTokens.length > 1
+      case 'swap-deposit':
+      case 'deposit':
+      default:
+        return availableInputTokens.length > 1
+    }
+  }, [mode, availableInputTokens, withdrawTokens, depositTokens])
```

Comment:
is this conditional needed? Don't we always populate the token selector using `availableInputTokens`? Can't we just check `availableInputTokens` for all cases?